### ISSUE TITLE
No newline after expression body function made with `addCode`

### ISSUE THREAD

**Issue by enjoydambience** [square/kotlinpoet#947]
This one's rather specific.
When:
- A property has a custom getter of the form `return xxx` and so can be an expression body
- It was added using the `addCode` builder function (instead of `addStatement`)

It does not insert a newline after the getter and may result in generating non-compiling code.

This does not happen if the getter is not converted to an expression body function.

It should either throw an error or correct itself by adding a newline.

Example:
```kotlin
val prop = PropertySpec.builder("foo", Int::class)
    .mutable()
    .getter(FunSpec.getterBuilder()
            .addCode("return 0")
            .build())
    .setter(FunSpec.setterBuilder()
            .addParameter("value", Int::class)
            .build())
    .build()
```
produces, verbatim:
```kotlin
var foo: kotlin.Int
  get() = 0set(value) {
  }
```

**Comment by enjoydambience** [comment]
Same happens for regular functions:
```kotlin
val type = TypeSpec.classBuilder("Foo")
    .addFunction(
        FunSpec.builder("foo")
            .returns(Int::class)
            .addCode("return 0")
            .build()
    ).build()
```
produces, verbatim:
```kotlin
class Foo {
  fun foo(): kotlin.Int = 0}
```

**Comment by JakeWharton** [comment]
To be clear, the proposal here is that we should always add a trailing newline to a function's `CodeBlock` even if one isn't provided? That would fix these cases, and the general case where you simply omit it.

Alternatively, the current behavior seems to be working-as-intended since `addCode` puts the responsibility of newlines after statements on you. Is there a reason you're using `addCode` and not `addStatement`?

**Comment by enjoydambience** [comment]
Digging through the code, it seems that trailing newlines are already ensured on functions if it is _not_ an expression body. This behavior shouldn't change if the function is an expression body.

(The fix is easy too; I have a PR ready)

**Comment by enjoydambience** [comment]
I discovered this when experimenting with generating properties. Using `addCode` for small functions wasn't an issue before.

**Comment by enjoydambience** [comment]
If this gets an "OK" I'll make a PR

**Comment by Egorand** [comment]
> Digging through the codebase, it seems that trailing newlines are already ensured on functions if it is not an expression body. This behavior shouldn't change if the function is an expression body.

Agree!

**Comment by Egorand** [comment]
Fixed by #948
